autograd/numpy/fft.py

```diff
@@ -44,3 +71,51 @@ def check_no_repeated_axes(*args, **kwargs):
         except (TypeError, KeyError):
             # no iterable axes argument
             pass
+
+def check_even_shape(axes, shape):
+    if shape[axes[-1]] % 2 != 0:
+        raise NotImplementedError("Real FFT gradient for odd lengthed last axes is not implemented.")
+
+def find_axes(nd, n, *args, **kwargs):
```

Comment:
The default value handling is according to their documentation.
In the end here the approach is to normalize the input as rfftn's, such that the body of the norm and factor calculation remains the same. I will give your suggestion a try, but gut feeling is it will look more abstract -- because now the main logic will have a function pointer of the underlining transform.